service/openapi/src/main/java/org/dotwebstack/framework/service/openapi/query/GraphQlQueryBuilder.java

```diff
@@ -1,51 +1,53 @@
 package org.dotwebstack.framework.service.openapi.query;
 
 import static org.dotwebstack.framework.core.helpers.ExceptionHelper.invalidConfigurationException;
-import static org.dotwebstack.framework.core.helpers.TypeHelper.getTypeString;
-import static org.dotwebstack.framework.service.openapi.response.ResponseContextHelper.getPathsForSuccessResponse;
-import static org.dotwebstack.framework.service.openapi.response.ResponseContextHelper.isExpanded;
 
-import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
-import java.util.Objects;
 import java.util.Optional;
 import java.util.Set;
-import java.util.StringJoiner;
 import java.util.stream.Collectors;
 import lombok.NonNull;
-import org.dotwebstack.framework.core.query.GraphQlField;
+import org.dotwebstack.framework.core.InvalidConfigurationException;
+import org.dotwebstack.framework.service.openapi.query.model.Field;
+import org.dotwebstack.framework.service.openapi.query.model.GraphQlQuery;
+import org.dotwebstack.framework.service.openapi.response.DwsQuerySettings;
 import org.dotwebstack.framework.service.openapi.response.ResponseSchemaContext;
+import org.dotwebstack.framework.service.openapi.response.ResponseTemplate;
 
 public class GraphQlQueryBuilder {
 
   public Optional<String> toQuery(@NonNull ResponseSchemaContext responseSchemaContext,
       @NonNull Map<String, Object> inputParams) {
 
-    if (responseSchemaContext.getGraphQlField() == null) {
+    DwsQuerySettings dwsQuerySettings = responseSchemaContext.getDwsQuerySettings();
+    String queryName = dwsQuerySettings.getQueryName();
+
+    if (queryName == null || queryName.isEmpty()) {
       return Optional.empty();
     }
 
-    Set<String> requiredPaths = getPathsForSuccessResponse(responseSchemaContext, inputParams);
+    ResponseTemplate okResponse = responseSchemaContext.getResponses()
+        .stream()
+        .filter(r -> r.getResponseCode() == 200)
+        .findFirst()
+        .orElseThrow(() -> new InvalidConfigurationException("No OK response found"));
 
-    var builder = new StringBuilder();
-    var joiner = new StringJoiner(",", "{", "}");
-    var argumentJoiner = new StringJoiner(",");
+    List<Field> fields = OasToGraphQlHelper.toQueryFields(okResponse, inputParams);
+    return toQuery(queryName, fields);
+  }
 
-    Set<String> queriedPaths = new HashSet<>();
-    addToQuery(responseSchemaContext.getGraphQlField(), requiredPaths, queriedPaths, joiner, argumentJoiner,
-        inputParams, true, "");
-    validateRequiredPathsQueried(requiredPaths, queriedPaths);
+  private Optional<String> toQuery(String queryName, List<Field> nodes) {
```

Comment:
```suggestion
  private Optional<String> toQuery(String queryName, List<Field> fields) {
```
Zo heet het twee regels erboven ook